dbclient/WorkspaceClient.py

```diff
@@ -397,6 +408,8 @@ def log_acl_to_file(self, artifact_type, read_log_filename, write_log_filename,
                     continue
                 acl_resp.pop('http_status_code')
                 write_fp.write(json.dumps(acl_resp) + '\n')
+                write_fp.flush()
```

Comment:
2 questions:

1. I am worried if we do the flushing on every key, the performance can be visibly degraded. Is this something we should worry about?

2. Since the flush will just make sure flushing the python internal buffered data to the os buffered data, are we stating that we are just guarding against the Python app crash scenario, rather than the whole OS system crashing?

Reply:
Results from running jobs:
**Original**
Complete Workspace Export Time: 0:00:02.802726
Complete Notebook ACLs Export Time: 0:00:04.213943
Complete Directories ACLs Export Time: 0:00:02.950634

**New**
Complete Workspace Export Time: 0:00:02.884954 (negligible change)
Complete Notebook ACLs Export Time: 0:00:04.948932 (+15%)
Complete Directories ACLs Export Time: 0:00:03.211578 (+10%)

Reply:
@kevinkim9264 the performance is minimally impacted with the flush

Reply:
Re Kevin:

1. I think the performance degradation due to flushing is eligible because we only choose to checkpoint slow operations which should take at least dozens of seconds or minutes.
2. It's a good point. I think we should call os.fsync to guarantee the correctness in any scenarios. We can wrap flush and fsync in one function, so there is no extra complexity. Also, performance penalty should be acceptable because disk sync takes ~10ms according to https://gist.github.com/jboner/2841832.

Reply:
@souravkhandelwal09 thank you for the benchmark result report!

@yubing 
1. I assume you meant negligible. And I agree that it's negligible if we indeed choose to checkpoint only for the heavy operations and just do the checkpoint at the stage level for the small operation ones. 
2. Yeah, though the chance is super super small for OS to crash and the checkpoint disk write to happen before the data save happen, if we are targeting 100% confidence level, we should consider fsyncing. With that said, this is only eligible if we are indeed checkpointing only for the heavy operations. If we fsync on every key operations, I believe the performance degradation will be an issue.